### PR TITLE
Diagnostics GAX V4

### PR DESCRIPTION
**Pull request by amanda-tarafa**

### CODE REVIEW

File: apis/apis.json [hunk 1]
```diff
@@ -1130,12 +1130,13 @@
...
         "Google.Cloud.Trace.V1": "2.3.0",
-        "Microsoft.Extensions.Http": "2.1.1",
+        "Microsoft.Extensions.Http": "6.0.0",
         "System.Diagnostics.StackTrace": "4.3.0"
       },
       "testDependencies": {
         "Google.Api.Gax.Testing": "3.7.0",
-        "Google.Cloud.ErrorReporting.V1Beta1": "2.0.0-beta05"
+        "Google.Cloud.ErrorReporting.V1Beta1": "2.0.0-beta05",
+        "Microsoft.Extensions.Hosting": "6.0.0"
```
Comment (jskeet) [inline apis/apis.json:1139]: Is this needed as a test dependency as well? I would have expected it to just be fine in the production project.

File: detect-pr-changes.sh [hunk 2]
```diff
@@ -67,16 +67,19 @@ done
 
 echo ""
 
+releasedApis=$apis
```
Comment (jskeet) [inline detect-pr-changes.sh:70]: A possibly-simpler option would be to create a new empty array, and add to it in the "else" block. But this is fine.

**Review by jskeet — APPROVED** [review]
So much nicer :)

File: apis/apis.json (hunk 1 above)
Comment (amanda-tarafa) [reply apis/apis.json:1139]: In production we have Extensions.Http and this is Extensions.Hosting :)

File: apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/ErrorReporting/ErrorReportingOptions.cs [hunk 3]
```diff
@@ -62,15 +48,15 @@ private ErrorReportingOptions(string logName, MonitoredResource resource, Buffer
             RetryOptions = GaxPreconditions.CheckNotNull(retryOptions, nameof(retryOptions));
         }
 
+        // TODO: Consider making this obsolete and creating a new Create now that we don't have the overload ambiguity.
```
Comment (amanda-tarafa) [inline apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/ErrorReporting/ErrorReportingOptions.cs:51]: @jskeet Flagging this TODO.
I say we go for it, even if we start up the new version with some obsolete code.

Comment (jskeet) [reply apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/ErrorReporting/ErrorReportingOptions.cs:51]: Yes - or just rename it to Create entirely? I don't mind.

Comment (amanda-tarafa) [reply apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/ErrorReporting/ErrorReportingOptions.cs:51]: Renamed entirely on the last commit. I'll merge on green.

**Pull request merged**